tests/framework/utils_imdsv2.py

```diff
@@ -0,0 +1,72 @@
+"""
+A simple IMDSv2 client
+
+- https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html
+- https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-categories.html
+"""
+
+import sys
+import time
+
+import requests
+
+class IMDSv2Client:
+    IMDSV2_HEADER_TOKEN_TTL = "X-aws-ec2-metadata-token-ttl-seconds"
+    IMDSV2_HEADER_TOKEN = "X-aws-ec2-metadata-token"
+
+    def __init__(self, endpoint="http://169.254.169.254/latest"):
```

Comment:
What does this IP address point to? Will it always be available? If this points at a test instance whose address may change, maybe a DNS resource needs to be synthesized that we can route requests to that will always have an instance behind it to process the request?

Reply:
This is the IMDS address. https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html

Reply:
Ah I see thanks. might be worth another named constant just for easier reading.